### ISSUE TITLE
Indication to identify protected and private methods

### ISSUE THREAD

**Issue by noraj** [lsegal/yard#1372]
`--protected` and `--private` of `yard doc` show protected and private methods on the documentation.
But once you have enabled them, you have public, protected and privates methods all mixed without a way to know which is what.
It would be nice to have a little badge next to protected and private methods to allow to identify them. 

## Expected result

Something like that

![image](https://user-images.githubusercontent.com/16578570/109015601-953ea180-76b5-11eb-9cee-41b9750f17c5.png)

![image](https://user-images.githubusercontent.com/16578570/109016042-0bdb9f00-76b6-11eb-93a4-36b13a6ff026.png)

**Comment by lsegal** [comment]
It doesn't seem like your methods are correctly being detected as private/protected, YARD should already be showing these tags. This is the output for the following minimal repro:

```ruby
class A
  protected

  def a; end
  def b; end

  private

  def c; end
  def d; end
end
```

![image](https://user-images.githubusercontent.com/686/109056511-5d276680-7695-11eb-831e-15b64b4a6dc8.png)

**Comment by noraj** [comment]
Yes I can se the CSS `public` class set on each `<li>`. I also see that without `--protected` or `--private` my method still appears in yard.
I didn't use the following syntax:

```ruby
class A
  protected

  def a; end
  def b; end

  private

  def c; end
  def d; end
end
```

but instead this one:

```ruby
class A
  def a; end
  def b; end
  def c; end
  def d; end

  protected :a, :b
  private :c, :d
end
```

Which is valid ruby and totally works as expected

![image](https://user-images.githubusercontent.com/16578570/109130893-8954ed00-7752-11eb-9ce6-157ed3e0c289.png)

Also I don't know if it matters but I'm in `class << self` because `class << self` + `private` is similar to `private_class_method` and I saw there  #1162 yard had issues with inline `private_class_method` but you said as a workaround that `private_class_method` was working with yard if used in a separate line. this may be different and #1162 is from 2018 but I mention it as it may lead somewhere.

So I have

```ruby
class A

  class << self
    def a; end
    def b; end
    def c; end
    def d; end

    protected :a, :b
    private :c, :d
  end
end
```

**Comment by noraj** [comment]
I just tried

```ruby
class A

  class << self
    def a; end
    def b; end

    protected

    def c; end
    def d; end
  end
end
```

Only a & b are show (which confirms that the alternative syntax shown earlier is not supported by yard) but even using the "classic" syntax seems to be bugged inside class methods but adding `--protected` never show the methods c & d.

It seems that whatever the syntax is `potected_class_method` or `class << self` + "section" `protected` or `class << self` +  `protected c:, d:` yard has an issue with class methods that doesn't appears with instance methods.

**Comment by noraj** [comment]
I'm digging some old maybe related issues #847, #760, #551 (I stopped after because it was getting too old)

**Comment by noraj** [comment]
I'll me my repository public soon so you can see the real code.

**Comment by noraj** [comment]
Source: https://github.com/sec-it/pass-station
Yard doc: https://sec-it.github.io/pass-station/yard/PassStation.html

@lsegal I can confirms that protected badge is displayed only for instance methods and not for class methods.

**Comment by lsegal** [comment]
> [!NOTE]
> This is a batch message, apologies if this feels impersonal.

I am closing out a few stale issues and PRs in the YARD repo and this was flagged for automatic closure because the issue may have been fixed or is likely not planned due to the issue/PR age. If this is still an issue or you believe the issue/PR is still worth a discussion, please feel free to comment or reopen a new issue. Thanks and sorry for not getting around to this the first time!
